app/furniture/marketplace/delivery_area_component.rb

```diff
@@ -28,20 +28,30 @@ def edit_button?
       delivery_area.persisted? && policy(delivery_area).edit?
     end
 
+    def discard_button
+      return unless discard_button?
+
+      ButtonComponent.new(label: "#{t("icons.discard")} #{t("discard.link_to")}",
+        title: t("marketplace.delivery_areas.discard.link_to", name: delivery_area.label),
+        href: delivery_area.location, method: :delete, scheme: :secondary)
+    end
+
+    def discard_button?
```

Comment:
Minor: it feels like this method and the `destroy_button?` methods want to instead to be methods on Delivery Area, `DeliveryArea#discardable?` and `DeliveryArea#destroyable?`. Then those can be unit-tested in the model spec, and this component calls those methods to determine if the buttons can/should be displayed.

It also helps the model self-document its behavior in terms of when it can be discarded or destroyed.

Reply:
I dig it! I've left the policy check but pulled out the state checks.

Reply:
I was thinking the same thing @anaulin! 

Was thinking about how in React (or similar) -- which supposedly influences ViewComponent -- it wasn't encouraged to pass entities into a component but instead a _view_ over the model<sup>1</sup> with strong indirections already prepped so you wouldn't end up coupling your entity layer to your view layer so tightly. In that modality we'd have a view function like: 

```ts
class MyComponent {
	constructor(data) {
		this.model = data;
	}

	showDiscardButton() {
		return this.model.canRemoveDeliveryArea;
	}
}
```ts

<sup>1</sup> Presenter, view model, etc... but a simple data bag